modules/500-upmeter/images/upmeter/src/pkg/probe/checker/d8_clusterconfiguration.go

```diff
@@ -263,3 +285,77 @@ func (c *checkMirrorValueChecker) Check() check.Error {
 	}
 	return nil
 }
+
+type poller interface {
+	Poll() (*status, check.Error)
+}
+
+type poll struct {
+	access    kubernetes.Access
+	namespace string
+	svcname   string
+	client    http.Client
+}
+
+func newPoller(access kubernetes.Access, namespace string, svcname string, client http.Client) poller {
+	return &poll{
+		access:    access,
+		namespace: namespace,
+		svcname:   svcname,
+		client:    client,
+	}
+}
+
+type status struct {
+	ConvergeInProgress        int  `json:"CONVERGE_IN_PROGRESS"`
+	ConvergeWaitTask          bool `json:"CONVERGE_WAIT_TASK"`
+	StartupConvergeDone       bool `json:"STARTUP_CONVERGE_DONE"`
+	StartupConvergeInProgress int  `json:"STARTUP_CONVERGE_IN_PROGRESS"`
+	StartupConvergeNotStarted bool `json:"STARTUP_CONVERGE_NOT_STARTED"`
+}
+
+func (c *poll) Poll() (*status, check.Error) {
+	url, err := c.extractServiceURL()
+	if err != nil {
+		return nil, check.ErrUnknown("cannot get svc url %s: %v", c.namespace, err)
+	}
+
+	req, err := http.NewRequestWithContext(context.TODO(), http.MethodGet, url, nil)
+	if err != nil {
+		return nil, check.ErrUnknown("error getting deckhouse pod status converge %s: %v", c.namespace, err)
+	}
+
+	resp, err := c.client.Do(req)
+	if err != nil || resp.StatusCode != http.StatusOK {
+		return nil, check.ErrUnknown("failed to get status converge %s: %v", c.namespace, err)
+	}
+
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, check.ErrUnknown("failed to get status converge %s: %v", c.namespace, err)
+	}
+
+	var status status
+	if err := json.Unmarshal(body, &status); err != nil {
+		return nil, check.ErrUnknown("failed unmarshal json %s: %v", c.namespace, err)
+	}
+
+	return &status, nil
+}
+
+func (c *poll) extractServiceURL() (string, error) {
+	service, err := c.access.Kubernetes().CoreV1().Services(c.namespace).Get(context.TODO(), svcName, metav1.GetOptions{})
```

Comment:
The client GETs the service by name whereas in templates the agent role is modified to LIST services. That's a mismatch that should be addressed.

Reply:
changed